### ISSUE TITLE
optimize: Test_NNLS.test_nnls and TestNNLS.test_maxiter fails with missing n

### ISSUE THREAD

**Issue by JS-Mechanic** [scipy/scipy#12273]
**Test_NNLS.test_nnls** and **TestNNLS.test_maxiter**
![Failed_tests](https://user-images.githubusercontent.com/57490926/83322047-d78f9b00-a269-11ea-9854-9533e9f2717f.png)

**Comment by EverLookNeverSee** [comment]
@ilayn 
Do you have an idea about how can we fix this?

**Comment by ilayn** [comment]
That's strange. I can't check right now but let's ping some optimize residents @mdhaber @Kai-Striega and others

**Comment by mdhaber** [comment]
In what context is this occurring?

**Comment by EverLookNeverSee** [comment]
I think it should occur here:
https://github.com/scipy/scipy/blob/0a7bc723d105288f4b728305733ed8cb3c8feeb5/scipy/optimize/_nnls.py#L80

**Comment by mdhaber** [comment]
I mean on what platform and with what version of SciPy, Python, etc. Does it happen in master?

**Comment by Kai-Striega** [comment]
It seems to work for my system (WSL-Ubuntu) on master

```python
Building, see build.log...
Build OK (0:00:23.545716 elapsed)
============================= test session starts ==============================
platform linux -- Python 3.6.9, pytest-5.4.3, py-1.8.2, pluggy-0.13.1 -- /home/kai/pyenvs/scipy-dev/bin/python
cachedir: .pytest_cache
rootdir: /mnt/c/Users/Kai/Develop/scipy, inifile: pytest.ini
plugins: forked-1.1.3, xdist-1.32.0
collecting ... collected 2 items

scipy/optimize/tests/test_nnls.py::TestNNLS::test_nnls PASSED            [ 50%]
scipy/optimize/tests/test_nnls.py::TestNNLS::test_maxiter PASSED         [100%]

============================== 2 passed in 0.28s ===============================

 NNLS quitting on iteration count.
```

**Comment by EverLookNeverSee** [comment]
> I mean on what platform and with what version of SciPy, Python, etc. Does it happen in master?

platform --> manjaro linux
python version --> 3.8.3
scipy version --> scipy development version
numpy --> 1.18.1

**Comment by mdhaber** [comment]
Hmm same for me on Windows, Python 3.7.5, Numpy 1.17.2. Any guesses on why it's not failing in CI?

**Comment by mnmelo** [comment]
I'm having this error in version 1.5.2, installed via pip (on Ubuntu 18.04) when following the exact same steps as in the docs:
```python3
import numpy as np
from scipy.optimize import nnls

A = np.array([[1, 0], [1, 0], [0, 1]])
b = np.array([2, 1, 1])

nnls(A, b)
> TypeError: _nnls.nnls() missing required argument 'n' (pos 3)
```
I traced it down to filename/function-name clash since the move of some of the modules to a private namespace (d1c1191). In this case, the `__init__.py` of optimize is importing function `nnls` from the fortran lib directly (`__nnls.cpython-36m-x86_64-linux-gnu.so` in my case) instead of from the wrapper `_nnls.py`.

Also, for some reason, my install still has a non-private `nnls.py` hanging around, with mostly the same content as `_nnls.py` (which may explain why some contexts are able to work?).

As a quick workaround I just pointed `__init__.py` to import from the non-private `nnls.py` (`from .nnls import nnls`) but a cleaner solution is probably to rename the fortran library to some non-clashing name.

Issue #12619, for all its lack of info, maybe reflects the same problem.

Hope this helps.
